src/utils/cosmosdb_model_versioned.ts

```diff
@@ -71,6 +73,26 @@ export const generateVersionedModelId = <T, ModelIdKey extends keyof T>(
 export const incVersion = (version: NonNegativeInteger): NonNegativeInteger =>
   (Number(version) + 1) as NonNegativeInteger;
 
+const getPartitionKeyFromSearchKey = <
+  T,
+  ModelIdKey extends keyof T,
+  PartitionKey extends keyof T = ModelIdKey
+>(
+  searchKey: DocumentSearchKey<T, ModelIdKey, PartitionKey>
+): CosmosPartitionKey => {
+  const builder = new PartitionKeyBuilder();
```

Comment:
why do we need these changes? I read "If you are not using Hierarchical Partitioning feature, Definition of Partition Key is practically backward compatible" here:

https://github.com/Azure/azure-sdk-for-js/blob/main/sdk/cosmosdb/cosmos/CHANGELOG.md

Reply:
because our `partitionKey` has this type annotation:

```ts
> = (T & BaseModel)[ModelIdKey] extends string // narrow type to the ones that might be an identity
  ? PartitionKey extends ModelIdKey // eslint-disable-next-line functional/prefer-readonly-type
    ? [(T & BaseModel)[ModelIdKey]] // eslint-disable-next-line functional/prefer-readonly-type
    : PartitionKey extends keyof (T & BaseModel) // eslint-disable-next-line functional/prefer-readonly-type
    ? [(T & BaseModel)[ModelIdKey], (T & BaseModel)[PartitionKey]]
    : never
  : never;
```

that is not compatible with the `PartitionKey` type defined by `@azure/cosmos`

Reply:
aaaaaaaaa! I understand now what you mean, I can remove the builder!